### ISSUE TITLE
.WithRedisCommander() produces broken commander instance

### ISSUE THREAD

**Issue by hades200082** [dotnet/aspire#5509]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

```cs
builder.AddRedis("redis")
    .WithDataVolume()
    .WithRedisCommander();
```

When loading Redis commander from the link in the aspire dashboard the UI seems unable to connect to the Redis instance.

### Expected Behavior

It should just work.

### Steps To Reproduce

Create a new AppHost with the code above and run it.

My AppHost has some other services registered:

```cs
var builder = DistributedApplication.CreateBuilder(args);

builder.AddKeycloak("keycloak")
    .WithDataVolume();

builder.AddRabbitMQ("rabbitmq")
    .WithDataVolume()
    .WithManagementPlugin();

builder.AddRedis("redis")
    .WithDataVolume()
    .WithRedisCommander();

builder.AddPostgres("postgres")
    .WithDataVolume()
    .WithPgAdmin()
    .AddDatabase("database");

builder.Build().Run();
```

### .NET Version info

.NET SDK:
 Version:           8.0.401
 Commit:            811edcc344
 Workload version:  8.0.400-manifests.57f7c351
 MSBuild version:   17.11.4+37eb419ad

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.22631
 OS Platform: Windows
 RID:         win-x64
 Base Path:   C:\Program Files\dotnet\sdk\8.0.401\

.NET workloads installed:
Configured to use loose manifests when installing new manifests.
 [wasm-tools]
   Installation Source: VS 17.10.35201.131
   Manifest Version:    8.0.8/8.0.100
   Manifest Path:       C:\Program Files\dotnet\sdk-manifests\8.0.100\microsoft.net.workload.mono.toolchain.current\8.0.8\WorkloadManifest.json
   Install Type:              Msi

 [aspire]
   Installation Source: VS 17.10.35201.131
   Manifest Version:    8.2.0/8.0.100
   Manifest Path:       C:\Program Files\dotnet\sdk-manifests\8.0.100\microsoft.net.sdk.aspire\8.2.0\WorkloadManifest.json
   Install Type:              Msi

Host:
  Version:      8.0.8
  Architecture: x64
  Commit:       08338fcaa5

.NET SDKs installed:
  6.0.201 [C:\Program Files\dotnet\sdk]
  6.0.321 [C:\Program Files\dotnet\sdk]
  7.0.203 [C:\Program Files\dotnet\sdk]
  8.0.108 [C:\Program Files\dotnet\sdk]
  8.0.206 [C:\Program Files\dotnet\sdk]
  8.0.304 [C:\Program Files\dotnet\sdk]
  8.0.401 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.3 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.26 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.33 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.20 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 8.0.6 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 8.0.8 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.3 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.12 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.26 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.33 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.20 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 8.0.6 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 8.0.8 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 6.0.3 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.26 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.33 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 7.0.5 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 7.0.20 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 8.0.6 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 8.0.8 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

Other architectures found:
  x86   [C:\Program Files (x86)\dotnet]
    registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\x86\InstallLocation]

Environment variables:
  Not set

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download

### Anything else?

Installed Workload Id      Manifest Version      Installation Source
--------------------------------------------------------------------
wasm-tools                 8.0.8/8.0.100         VS 17.10.35201.131
aspire                     8.2.0/8.0.100         VS 17.10.35201.131

**Comment by davidfowl** [comment]
@mitchdenny we made changes to use the new event model in 8.2, can you look into this?

**Comment by davidfowl** [comment]
I couldn't reproduce this on Windows with docker desktop.

**Comment by mitchdenny** [comment]
Yeah I just tried as well. Postgres is working also (I used TestShop). I'm going to try 8.2 rather than our repo just to be sure.

**Comment by mitchdenny** [comment]
Yeah can't repro on a relatively clean environment with 8.2. @hades200082 did this work for you with Aspire 8.1? Just wonder if there is something environmental going on.

**Comment by hades200082** [comment]
@mitchdenny yes. Seemed to stop working after I updated to 8.2

**Comment by mitchdenny** [comment]
OK. Can you confirm that the `redis-commander` container has a `REDIS_HOSTS` variable populated? It should look something like this (port will be different):

```env
REDIS_HOSTS=redis:host.docker.internal:50902:0
```

What error is Redis Commander showing you (or is it just case of you not being able to bring up Redis Commander at all?

**Comment by hades200082** [comment]
@mitchdenny I did a screencast of the issue using a brand new, bare-bones solution here: https://app.screencast.com/RP3tjbKvyQUc3

This covers this issue and the one reported in #5510 

I have two machines I've been seeing this issue on. One is my home Desktop PC which is an AMD-based (Ryzen 7) Windows 11 PC running Podman Desktop. The other is my work laptop which is an Intel-based (i7) Windows 10 Dell laptop running Docker Desktop.

Coming back to my work laptop after the weekend, it appears to have started working again here. It was failing on Friday :/

On my home PC (I'm WFH today) I still see this issue and the `REDIS_HOSTS` appears to be set correctly in the redis commander service according to the aspire dashboard.

I see no errors in the devtools console or network tabs on the redis commander page. Only the "Error: Status: Reconnecting" message as shown in the screencast linked above.

**Comment by mitchdenny** [comment]
OK thanks for that. I've managed to be able to reproduce it when I switched the container runtime to podman. I don't think its a Redis Commander or PGAdmin specific issue, I suspect its a container to container networking/name resolution issue.

What is strange is that you said that you are also seeing it on another machine that is running Docker Desktop (which I cannot reproduce). If I swap from Docker Desktop to Podman on the one machine the problem starts appearing, when I switch back it disappears.

**Comment by hades200082** [comment]
@mitchdenny It was definitely happening on my work laptop last week, which uses Docker Desktop, but I'm not able to recreate it today.

The only thing that's changed today is that I updated the .NET SDK version to the latest which would have updated the .NET runtime on my laptop from 8.0.3 to 8.0.8.

I also updated the .NET SDK on my home machine over the weekend so that is also running 8.0.8 and still has the issue using Podman.

**Comment by hades200082** [comment]
Another thought on this - Why is the RabbitMQ `.WithManagementPlugin()` unaffected, while RedisCommander and PGAdmin are both affected?

```cs
builder
    .AddRabbitMQ(options.ConnectionStringName, password: builder.AddParameter("AmqpPassword", true))
    .WithHealthCheck()
    .WithManagementPlugin()
    .WithImage("masstransit/rabbitmq") // must come after `.WithManagementPlugin()`
    .WithDataVolume()
```

**Comment by mitchdenny** [comment]
> Another thought on this - Why is the RabbitMQ `.WithManagementPlugin()` unaffected, while RedisCommander and PGAdmin are both affected?

The RabbitMQ plugin runs inside the main container.

**Comment by NiklasEi** [comment]
I can reproduce this on Linux (Ubuntu 24.04) running docker 27.2.0 and Aspire 8.2

**Comment by davidfowl** [comment]
Closing this one assuming it'll be fixed by #5584
